apps/tutorial_servo_control/src/TutorialServo.cpp

```diff
@@ -0,0 +1,31 @@
+#include "TutorialServo.h"
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees, float minPulsewidthInMs , float maxPulsewidthInMs){
+        m_servoRangeInDegrees =  servoRangeInDegrees;
+        m_minPulsewidthInMs = minPulsewidthInMs;
+	m_maxPulsewidthInMs = maxPulsewidthInMs;
+}
+
+float TutorialServo::getServoRangeInDegrees() const{
+
+        return (m_servoRangeInDegrees);
+}
+
+float TutorialServo::getMinPulseWidthInMs()const{
+
+        return (m_minPulsewidthInMs);
+
+}
+float TutorialServo::getMaxPulseWidthInMs()const{
+
+        return (m_maxPulsewidthInMs);
+
+}
+void TutorialServo::setPositionInDegrees(float degrees){
+
+         float slope{0.0}, pulse{0.0}, range{m_maxPulsewidthInMs - m_minPulsewidthInMs};
```

Comment:
try to avoid the declaration of intermediate variables. this is a simple enough calculation and you shouldnt need these 3 intermediate variables. just do it inside the funciton